script/releases/v0.1.0-rewardsv2/1-eoa.s.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import {EOADeployer} from "zeus-templates/templates/EOADeployer.sol";
+import {RewardsCoordinator} from "src/contracts/core/RewardsCoordinator.sol";
+import {IDelegationManager} from "src/contracts/interfaces/IDelegationManager.sol";
+import {DelegationManager} from "src/contracts/core/DelegationManager.sol";
+import {StrategyManager} from "src/contracts/core/StrategyManager.sol";
+import {EigenLabsUpgrade} from "../EigenLabsUpgrade.s.sol";
+import {Test, console} from "forge-std/Test.sol";
+
+contract Deploy is EOADeployer {
+    using EigenLabsUpgrade for *;
+
+    function _deploy() internal override {
+        vm.startBroadcast();
+        
+        deploySingleton(
+            address(
+                new RewardsCoordinator(
+                    IDelegationManager(zDeployedProxy(type(DelegationManager).name)),
+                    StrategyManager(zDeployedProxy(type(StrategyManager).name)),
+                    zUint32("REWARDS_COORDINATOR_CALCULATION_INTERVAL_SECONDS"),
+                    zUint32("REWARDS_COORDINATOR_MAX_REWARDS_DURATION"),
+                    zUint32("REWARDS_COORDINATOR_MAX_RETROACTIVE_LENGTH"),
+                    zUint32("REWARDS_COORDINATOR_MAX_FUTURE_LENGTH"),
+                    zUint32("REWARDS_COORDINATOR_GENESIS_REWARDS_TIMESTAMP")
+                )
+            ),
+            this.impl(type(RewardsCoordinator).name)
+        );
+
+        vm.stopBroadcast();
+    }
+
+    function testDeploy() public {
```

Comment:
There should be an additional check/ expect revert to ensure that the the `initialize` function can't be called in the implementation contract since it'll be disabled.